[Enhancement] The FE&BE&CN startup script automatically detects that the startup is successful

zhuxt2015 opened the pull request:
## Why I'm doing:

## What I'm doing:

Fixes #48369 

## What type of PR is this:

- [ ] BugFix
- [ ] Feature
- [x] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

Does this PR entail a change in behavior?

- [ ] Yes, this PR will result in a change in behavior.
- [x] No, this PR will not result in a change in behavior.

If yes, please specify the type of change:

- [ ] Interface/UI changes: syntax, type conversion, expression evaluation, display information
- [ ] Parameter changes: default values, similar parameters but with different default values
- [ ] Policy changes: use new policy to replace old one, functionality automatically enabled
- [ ] Feature removed
- [ ] Miscellaneous: upgrade & downgrade compatibility, etc.

## Checklist:

- [ ] I have added test cases for my bug fix or my new feature
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function
- [x] This is a backport pr

## Bugfix cherry-pick branch check:
- [x] I have checked the version labels which the pr will be auto-backported to the target branch
  - [x] 3.3
  - [x] 3.2
  - [x] 3.1
  - [x] 3.0
  - [x] 2.5

zhuxt2015 committed:
Add a hint that the process was successfully started

zhuxt2015 committed:
Add a hint that the process was successfully started

stephen-shelby committed:
[Feature] support iceberg query with branch/tag/version/timestamp (#47997)

Signed-off-by: stephen <stephen5217@163.com>

HangyuanLiu committed:
[BugFix] Fixed the bug that clusterid cannot be parsed when using --helper during upgrade (#48197)

Signed-off-by: HangyuanLiu <460660596@qq.com>

stephen-shelby committed:
[BugFix] fix npe on iceberg v1 table when table uuid is null (#48363)

Signed-off-by: stephen <stephen5217@163.com>

DanRoscigno committed:
[Doc] SEO for docker (#48392)

Updates to 8 of the most searched for pages.

Signed-off-by: DanRoscigno <dan@roscigno.com>

kevincai reviewed (changes requested):
Let's have a discussion before making this change.

Commit:
[Doc] Fix broken image in README (#48353)

Signed-off-by: 絵空事スピリット <wanglichen@starrocks.com>

stephen-shelby committed:
[Feature] Support specifying branch when writing to table (#48061)

Signed-off-by: stephen <stephen5217@163.com>

silverbullet233 committed:
[Enhancement] add enable_experimental_temporary_table configuration (#48362)

Signed-off-by: silverbullet233 <3675229+silverbullet233@users.noreply.github.com>

Commit:
[Doc] Update the default value of lake_autovacuum_grace_period_minutes (#48371)

Signed-off-by: 絵空事スピリット <wanglichen@starrocks.com>

packy92 committed:
[BugFix] fix not parsing prepare parameter in order (#48386)

Signed-off-by: packy92 <wangchao@starrocks.com>

Youngwb committed:
[Enhancement] upgrade delta kernel to 4.0rc1  (#48394)

Signed-off-by: Youngwb <yangwenbo_mailbox@163.com>

murphyatwork committed:
[BugFix] change the cputime in profile from max to sum calculation (#48370)

Signed-off-by: Murphy <mofei@starrocks.com>

silverbullet233 committed:
[Enhancement] fix memory counting issue in distinct agg state (#48195)

Signed-off-by: silverbullet233 <3675229+silverbullet233@users.noreply.github.com>

starrocks-xupeng committed:
[Enhancement] create star mgr dir on startup in shared-data cluster (#48382)

create starmgr dir when FE first starts up

Signed-off-by: starrocks-xupeng <xupeng@starrocks.com>

ZiheLiu committed:
[UT] Fix unstable FE UT QueryDumpActionTest (#48413)

Signed-off-by: zihe.liu <ziheliu1024@gmail.com>

zombee0 committed:
[BugFix]swallow error status when min/max type unsupported (#48300)

Signed-off-by: zombee0 <ewang2027@gmail.com>

murphyatwork committed:
[Enhancement] check window function of MV in subquery/cte (#48417)

Signed-off-by: Murphy <mofei@starrocks.com>

wyb committed:
[BugFix] Fix csv converter and chunk column inconsistent in select into outfile (#48052)

Signed-off-by: wyb <wybb86@gmail.com>

srlch committed:
[BugFix] Fix race condition between optimize table and archive task history (#48383)

Signed-off-by: srlch <linzichao@starrocks.com>

LiShuMing committed:
[Enhancement] (Multi Ref Base Table Part2)  MVPCTRefreshPlanBuilder supports multi ref base tables in mv refresh (#48284)

Signed-off-by: shuming.li <ming.moriarty@gmail.com>

sevev committed:
[BugFix] Fix some issue of add/drop field for struct column (#48126)

Signed-off-by: sevev <qiangzh95@gmail.com>

yingtingdong committed:
[Doc] Update Description in BE labels (#48372)

Signed-off-by: yingtingdong <737833890@qq.com>
Signed-off-by: 絵空事スピリット <wanglichen@starrocks.com>
Co-authored-by: 絵空事スピリット <wanglichen@starrocks.com>

Commit:
[Doc] Update 3.3.0 release notes (#48428)

Commit:
[Doc] Update the default value of Hive Catalog property (#48434)

Commit:
[Doc] Revise Arch and Helm QuickStart (#48426)

ZiheLiu committed:
[BugFix] Fix check of plan timeout in SeriallyTaskScheduler (#48405)

Signed-off-by: zihe.liu <ziheliu1024@gmail.com>

DanRoscigno committed:
[Doc] Update data_migration_tool.md en (#48462)

Signed-off-by: Dan Roscigno <dan@roscigno.com>

DanRoscigno committed:
[Doc] Update data_migration_tool.md (#48461)

Signed-off-by: Dan Roscigno <dan@roscigno.com>

sevev committed:
[Enhancement] Add memory check during pk apply (#47889)

Signed-off-by: sevev <qiangzh95@gmail.com>

xiangguangyxg committed:
[Enhancement] Add mv repair handler interface (#48365)

Signed-off-by: xiangguangyxg <xiangguangyxg@gmail.com>

wyb committed:
[Enhancement] Improve kafka authentication error message (#47649)

Signed-off-by: wyb <wybb86@gmail.com>

HangyuanLiu committed:
[BugFix] Fix getFullyQualifiedNameFromListAllowNull NPE when check system privilege bug (#48199)

Signed-off-by: HangyuanLiu <460660596@qq.com>

murphyatwork committed:
[Enhancement] simplify case-when true expression (#48419)

Signed-off-by: Murphy <mofei@starrocks.com>

murphyatwork committed:
[Feature] support lookup_string function (#48429)

Signed-off-by: Murphy <mofei@starrocks.com>

Astralidea committed:
[BugFix] Fix bug roll num does not work (#48493)

Signed-off-by: Astralidea <astralidea@163.com>

wyb committed:
[Enhancement] Improve kafka group authorization failed message in routine load (#46136)

Signed-off-by: wyb <wybb86@gmail.com>

Astralidea committed:
[Enhancement] BE remove unused zookeeper for CVEs (#48488)

Signed-off-by: Astralidea <astralidea@163.com>

DanRoscigno committed:
[Doc] Update regexp_extract.md (#48520)

Signed-off-by: Dan Roscigno <dan@roscigno.com>
Signed-off-by: 絵空事スピリット <wanglichen@starrocks.com>
Co-authored-by: 絵空事スピリット <wanglichen@starrocks.com>

ZiheLiu committed:
[BugFix] Make Broadcast Join generate deterministic GRF (#48496)

Signed-off-by: zihe.liu <ziheliu1024@gmail.com>

decster committed:
[Enhancement] Add metrics for key number in KeyMgr (#48258)

Signed-off-by: Binglin Chang <decstery@gmail.com>

letian-jiang committed:
[BugFix] Fix jdbc predicate on LargeStringLiteral (#48506)

Signed-off-by: Letian Jiang <letian.jiang@outlook.com>

LiShuMing committed:
[BugFix] [UT] Ensure to refresh base table for cached mv partition mappings (#48507)

Signed-off-by: shuming.li <ming.moriarty@gmail.com>

gengjun-git committed:
[Enhancement] Reject rename column name of tables in database: _statistic_ (#42761)

Signed-off-by: gengjun-git <gengjun@starrocks.com>

banmoy committed:
[Doc] Fix wrong description about transaction stream load label requirement (#48548)

Signed-off-by: PengFei Li <lpengfei2016@gmail.com>

murphyatwork committed:
[Enhancement] avoid use dict_mapping function in lookup_string (#48543)

Signed-off-by: Murphy <mofei@starrocks.com>

HangyuanLiu committed:
[Enhancement] Refactor ThriftConnectionPool and ThriftRPCRequestExecutor to Unify thrift rpc request (#48292)

Signed-off-by: HangyuanLiu <460660596@qq.com>

LiShuMing committed:
[UT] Fix unstable mv sql tests (#48562)

Signed-off-by: shuming.li <ming.moriarty@gmail.com>

Seaven committed:
[Enhancement] support count distinct on constant semi-type (#48273)

Signed-off-by: Seaven <seaven_7@qq.com>

Commit:
[Doc] Add Behavior Changes to 3.2.9 (#48567)

Commit:
[Doc] Add Doc on Temporary Table (#48549)

wyb committed:
[BugFix] Fix jobName filter not work when getting routine load jobs (#47907)

Signed-off-by: wyb <wybb86@gmail.com>

stdpain committed:
[BugFix] Fix wrong plan in Colocate Group Execution (#48565)

Signed-off-by: stdpain <drfeng08@gmail.com>

luohaha committed:
[Enhancement] make report audit statistics in sink operator asynchronous (#48162)

Signed-off-by: luohaha <18810541851@163.com>

nshangyiming committed:
[Feature] Add build arch info when showing FE/BE version (#48572)

Signed-off-by: Dejun Xia <xiadejun@starrocks.com>

nshangyiming committed:
[Enhancement] Set python3 as the default python version when compiling (#48586)

Signed-off-by: Dejun Xia <xiadejun@starrocks.com>

xiangguangyxg committed:
[Enhancement] Support getting physical partition by name (#47270)

Signed-off-by: xiangguangyxg <xiangguangyxg@gmail.com>

HangyuanLiu committed:
[BugFix] Fixed a bug where Compact and Schema Change could not be executed together (#48585)

Signed-off-by: HangyuanLiu <460660596@qq.com>

chenminghua8 committed:
[BugFix] Fix issue where range partition expr parameters were not checked (#48522)

Signed-off-by: chenminghua8 <cmptmn@126.com>

Commit:
[Doc] Add release notes 3.3.1 (#48597)

Signed-off-by: 絵空事スピリット <wanglichen@starrocks.com>

Smith-Cruise committed:
[Enhancement] Add exception stacktrace in orc reader to track exception (#48503)

Signed-off-by: Smith Cruise <chendingchao1@126.com>

nshangyiming committed:
[BugFix] Fix clang tidy compilation error (#48605)

Signed-off-by: Dejun Xia <xiadejun@starrocks.com>

Smith-Cruise committed:
[Enhancement] Support display datacache's hit rate in app level (#48450)

Signed-off-by: Smith Cruise <chendingchao1@126.com>

Astralidea committed:
[Enhancement] Broker Security Updates (#48616)

Signed-off-by: Astralidea <astralidea@163.com>

letian-jiang committed:
[Enhancement] Make ORC writer compatible with hive reader (#47777)

Signed-off-by: Letian Jiang <letian.jiang@outlook.com>

letian-jiang committed:
[BugFix] Case insensitive column name for iceberg sink (#47740)

Signed-off-by: Letian Jiang <letian.jiang@outlook.com>

stdpain committed:
[BugFix] Fix the problem that Stream/Routine Load memory statistics are not accurate. (#48587)

Signed-off-by: stdpain <drfeng08@gmail.com>

DanRoscigno committed:
[Doc] Helm translate (#48588)

murphyatwork committed:
[BugFix] fix the race condition of addPartition operation (#48492)

Signed-off-by: Murphy <mofei@starrocks.com>

Commit:
[Doc] Update release notes 3.3.0 (#48635)

Signed-off-by: 絵空事スピリット <wanglichen@starrocks.com>

dirtysalt committed:
[Refactor] remove unused `numInstances` field (#48608)

Signed-off-by: yanz <dirtysalt1987@gmail.com>

luohaha committed:
[Enhancement] avoid BE OOM when handle large number of tablet writes (#48495)

Signed-off-by: luohaha <18810541851@163.com>

yingtingdong committed:
[Doc] Shared-data supports partial update in column mode (#48612)

Signed-off-by: yingtingdong <737833890@qq.com>
Signed-off-by: 絵空事スピリット <wanglichen@starrocks.com>
Co-authored-by: 絵空事スピリット <wanglichen@starrocks.com>

dujijun007 committed:
[BugFix] Fix drop vector index bug (#48346)

Signed-off-by: leorishdu <18771113323@163.com>

sevev committed:
[BugFix] Fix some bug of alter struct column. (#48213)

Signed-off-by: sevev <qiangzh95@gmail.com>

tracymacding committed:
[BugFix] Fix 'PartitionKey' field of cloud native table if different with     shared nothing table when run 'show partitions' command (#48647)

Signed-off-by: tracymacding <tracymacding@gmail.com>

dujijun007 committed:
[Feature] Introduce TenAnn as the vector index implementation library (#48384)

Signed-off-by: leorishdu <18771113323@163.com>

wyb committed:
[UT] Fix lake compaction UT (#48648)

Signed-off-by: wyb <wybb86@gmail.com>

murphyatwork committed:
[BugFix] fix rpc failure error message (#48658)

Signed-off-by: Murphy <mofei@starrocks.com>

stdpain committed:
[BugFix] Fix Java UDAF OOM in spill or sorted streaming agg  (#48618)

Signed-off-by: stdpain <drfeng08@gmail.com>

HangyuanLiu committed:
[Enhancement] Move alter table clause op conflict check logic from AlterJobMgr to AlterTableStatementAnalyzer (#48603)

Signed-off-by: HangyuanLiu <460660596@qq.com>

HangyuanLiu committed:
[BugFix] It is strictly forbidden to use read-write locks inside Intend Lock (#48530)

Signed-off-by: HangyuanLiu <460660596@qq.com>

letian-jiang committed:
[BugFix] Fix several glue & s3 integration issues (#48433)

Signed-off-by: Letian Jiang <letian.jiang@outlook.com>

LiShuMing committed:
[Enhancement] (Multi Ref Base Table Part3) Supports to track multi ref base tables in partition compensate (#48192)

Signed-off-by: shuming.li <ming.moriarty@gmail.com>

luohaha committed:
[Enhancement] use process mem tracker when pk table preload state (#48661)

Signed-off-by: luohaha <18810541851@163.com>

sevev committed:
[Enhancement] Initialize olap_reader without copying tablet schema (#48485)

Signed-off-by: sevev <qiangzh95@gmail.com>

letian-jiang committed:
[Enhancement] Control parquet file size when general compression is applied (#47391)

Signed-off-by: Letian Jiang <letian.jiang@outlook.com>

xiangguangyxg committed:
[BugFix] Avoid hdfs fs manager interrupting the thread when exception occurs (#48403)

Signed-off-by: xiangguangyxg <xiangguangyxg@gmail.com>

stephen-shelby committed:
[Refactor] improve error log when creating iceberg view (#48673)

Signed-off-by: stephen <stephen5217@163.com>

stephen-shelby committed:
[Enhancement] improve log when getting table fails (#48676)

Signed-off-by: stephen <stephen5217@163.com>

stephen-shelby committed:
[BugFix] use default_catalog as catalog name when creating iceberg view does not use catalog name (#48681)

Signed-off-by: stephen <stephen5217@163.com>

stdpain committed:
[BugFix] Fix wrong result when enable spill preaggregation (#48700)

Signed-off-by: stdpain <drfeng08@gmail.com>

DanRoscigno committed:
[Doc] Rename competence_boundary-shared-data.md (#48705)

Signed-off-by: Dan Roscigno <dan@roscigno.com>
Signed-off-by: DanRoscigno <dan@roscigno.com>

DanRoscigno committed:
[Doc] move configuration up (#48716)

Signed-off-by: DanRoscigno <dan@roscigno.com>

Commit:
[Doc] Fix typos and misused formats (#48689)

Signed-off-by: 絵空事スピリット <wanglichen@starrocks.com>

wangsimo0 committed:
[Doc]Update FE_configuration.md (#48665)

switch to true since v 3.3.0 ga version

Signed-off-by: simo <48942089+wangsimo0@users.noreply.github.com>

wangsimo0 committed:
[Doc]Update FE_configuration.md (#48668)

Signed-off-by: simo <48942089+wangsimo0@users.noreply.github.com>

monchickey committed:
[Doc] Update the unit of new_planner_optimize_timeout variable in System_va… (#48654)

Signed-off-by: monchickey <75814968+monchickey@users.noreply.github.com>

wangsimo0 committed:
[Doc]Update release-3.3.md (#48624)

Signed-off-by: simo <48942089+wangsimo0@users.noreply.github.com>
Signed-off-by: 絵空事スピリット <wanglichen@starrocks.com>
Co-authored-by: 絵空事スピリット <wanglichen@starrocks.com>

wangsimo0 committed:
[Doc]Update release-3.3.md (#48625)

Signed-off-by: simo <48942089+wangsimo0@users.noreply.github.com>
Signed-off-by: 絵空事スピリット <wanglichen@starrocks.com>
Co-authored-by: 絵空事スピリット <wanglichen@starrocks.com>

Commit:
[Doc] SHOW PROC details (#48687)

Signed-off-by: 絵空事スピリット <wanglichen@starrocks.com>

DanRoscigno committed:
[Doc] rename assets dir (#48737)

Include files and images belong in a dir named _assets. Fixing an issue related to switch from Gatsby to Docusaurus.

Signed-off-by: DanRoscigno <dan@roscigno.com>

sevev committed:
[BugFix] Fix the bug in the persistent index that checks whether page reading is supported. (#48659)

Signed-off-by: sevev <qiangzh95@gmail.com>

eshishki committed:
[BugFix] make caching io closable (#47944)

yingtingdong committed:
[Doc] New module for audit_log_modules (#48751)

Signed-off-by: yingtingdong <737833890@qq.com>
Signed-off-by: 絵空事スピリット <wanglichen@starrocks.com>
Co-authored-by: 絵空事スピリット <wanglichen@starrocks.com>

meegoo committed:
[Refactor] Add max column table limit max_column_number_per_table for table (#47869)

Signed-off-by: meegoo <meegoo.sr@gmail.com>

Smith-Cruise committed:
[BugFix] Fix compile error without starcache lib (#48755)

Signed-off-by: Smith Cruise <chendingchao1@126.com>

gengjun-git committed:
[BugFix] Fix Colocate Group marked as stable when checking urgent tables (#48680)

Signed-off-by: gengjun-git <gengjun@starrocks.com>

kevincai committed:
[Enhancement] support json log4j logging (#48546)

Signed-off-by: Kevin Xiaohua Cai <caixiaohua@starrocks.com>

kaijianding committed:
[BugFix] Fix NPE due to fail to get rollup function name (#48142)

Signed-off-by: kaijian.ding <kaijian.ding@gmail.com>

murphyatwork committed:
[Enhancement] add trace timer for Lock (#48662)

Signed-off-by: Murphy <mofei@starrocks.com>

yandongxiao committed:
[UT] Fix unit test in LeaderOpExecutorTest.java (#48766)

Signed-off-by: yandongxiao <yandongxiao@starrocks.com>

wyb committed:
[BugFix] Fix pipe lost file (#48756)

Signed-off-by: wyb <wybb86@gmail.com>

zhuxt2015 committed:
Use the /api/health interface to determine that the process has been successfully started

zhuxt2015 committed:
Merge remote-tracking branch 'origin/start_script_improve' into start_script_improve

# Conflicts:
#	bin/start_backend.sh
#	bin/start_fe.sh
